ansible/configs/sap-integration/software.yml

```diff
@@ -14,104 +14,110 @@
   gather_facts: True
   tasks:
 
-    - name: Ensure NFS directory exists
-      file:
-        path: "/nfs"
-        state: directory
-
-    - name: Mount up device by UUID
-      mount:
-        path: /nfs
-        src: /dev/vdb
-        fstype: xfs
-        state: present
-
-    - name: 'Ensure required packages are installed'
-      package:
-        name: '{{ item }}'
-        state: installed
-      with_items:
-      - nfs-utils
-      - firewalld
-
-    - name: 'Ensure firewalld is running'
-      service:
-        name: firewalld
-        state: started
-        enabled: yes
-
-    - name: 'Open Firewall for NFS use'
-      firewalld:
-        port: "{{ item }}"
-        permanent: yes
-        state: enabled
-        immediate: yes
-      with_items:
-      - 111/tcp
-      - 111/udp
-      - 2049/tcp
-      - 2049/udp
-
-    - name: "Ensure export file contains the directory to be shared"
-      lineinfile:
-        path: /etc/exports
-        state: present
-        regexp: "^/nfs"
-        line: "/nfs *(insecure,rw,no_root_squash,no_wdelay,sync)"
-
-    - name: 'Ensure nfs-server is restarted and running'
-      service:
-        name: nfs-server
-        state: restarted
-        enabled: yes
-
-    - name: 'Ensure SFLIGHT export information can be read by anyone'
-      file:
-        path: /nfs/SFLIGHT/index
-        state: directory
-        recurse: yes
-        owner: root
-        group: root
-        mode: 0755
-
-    - name: Ensure /nfs directory is owned by 'cloud-user'
-      file:
-        path: /nfs
-        state: directory
-        owner: cloud-user
-
-    - name: Ensure SAP software is not present for external (non Red Hatters) requesters
-      file:
-        state: absent
-        path: "{{ item }}"
-      loop:
-        - '/nfs/HANA_installation/'
-        - '/nfs/S4HANA_installation/'
-        - '/nfs/SAPCAR/'
-        - '/nfs/SAPHOSTAGENT/'
-      when: not rh_internal
+    - name: OSP NFS specific tasks
+      when: cloud_provider is match("osp")
+      block:
```

Comment:
Please prefer `include_tasks` instead of `block` so the tasks are not evaluated or displayed in the logs in case the cloud provider is not osp